Library/Homebrew/prefix.sh

```diff
@@ -17,10 +17,10 @@ homebrew-prefix() {
   [ -z "$formula" ] && echo "$HOMEBREW_PREFIX" && return 0
 
   local formula_path
-  if [ -f "$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core/${formula}.rb" ]; then
-    formula_path="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core/${formula}.rb"
+  if [ -f "$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core/Formula/${formula}.rb" ]; then
+    formula_path="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core/Formula/${formula}.rb"
   else
-    formula_path="$(find "$HOMEBREW_REPOSITORY/Library/Taps" -name "${formula}.rb" -print -quit)"
+    formula_path="$(find "$HOMEBREW_REPOSITORY/Library/Taps" -name "${formula}.rb" \! -path "*/Casks/${formula}.rb" -print -quit)"
```

Comment:
I'd rather this specifically checked the `Formula` directory.

Currently things like `lib/test_bot.rb` also get picked up.

Reply:
OK.

Reply:
> I'd rather this specifically checked the `Formula` directory.

It can't. We support formulae in the root directory of taps.

Reply:
I did not realise we supported this. We should definitely be more restrictive than just blocking Casks though.

Reply:
I think it's a balance between performance and correctness, here, and I lean towards performance.

Reply:
Could perhaps refactor https://github.com/Homebrew/brew/blob/master/Library/Homebrew/items.sh to use the filters from there without using `sed` and ensuring we still use `-print` `-quit` for an early return.

Reply:
Actually, it turns out there's at least one other option...

Old (incorrect) find - it finds casks like `google-chrome`:
```
find "$HOMEBREW_REPOSITORY/Library/Taps" -name "${formula}.rb" -print -quit
```
New glob "find":
```
shopt -s nullglob
echo "$HOMEBREW_REPOSITORY/Library/Taps"/*/*/{Formula/,HomebrewFormula/,}"${formula}.rb" | head -n 1
```
New regex-based find:
```
find $HOMEBREW_REPOSITORY/Library/Taps \
  -type f \( \
    -regex "^$HOMEBREW_REPOSITORY/Library/Taps/[^/]*/[^/]*/Formula/${formula}\.rb\$" -o \
    -regex "^$HOMEBREW_REPOSITORY/Library/Taps/[^/]*/[^/]*/HomebrewFormula/${formula}\.rb\$" -o \
    -regex "^$HOMEBREW_REPOSITORY/Library/Taps/[^/]*/[^/]*/${formula}\.rb\$" \
  \) \
  -print -quit
```

I wrote a script to test and benchmark all three options [here](https://gist.github.com/d69997e104d558659213ee8cb64dfaac). Would be good if someone could review the `clear_cache` portion to:
1. verify that I did it the right way on Linux
2. suggest the equivalent for macOS (my search-fu failed me)
3. fixup the script with [2] and run it on macOS (I got too many statistical outlier and first-run warnings from `hyperfine` to trust the results)

On Linux, I got the following results:
| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `new_glob umka` | 36.3 ± 6.4 | 24.9 | 51.9 | 1.05 ± 0.27 |
| `old_find umka` | 34.4 ± 6.1 | 26.3 | 49.6 | 1.00 |
| `new_find umka` | 39.6 ± 4.9 | 31.9 | 50.5 | 1.15 ± 0.25 |
| `new_glob endgame` | 36.7 ± 5.3 | 28.8 | 51.1 | 1.07 ± 0.24 |
| `old_find endgame` | 354.0 ± 7.7 | 345.8 | 370.9 | 10.29 ± 1.85 |
| `new_find endgame` | 518.2 ± 37.1 | 481.0 | 592.2 | 15.07 ± 2.89 |
| `new_glob facter` | 41.0 ± 8.6 | 31.0 | 71.0 | 1.19 ± 0.33 |
| `old_find facter` | 346.5 ± 10.1 | 332.0 | 364.1 | 10.08 ± 1.82 |
| `new_find facter` | 475.4 ± 10.7 | 466.3 | 498.3 | 13.82 ± 2.48 |
| `new_glob google-chrome` | 37.3 ± 7.8 | 28.2 | 64.1 | 1.08 ± 0.30 |
| `old_find google-chrome` | 245.6 ± 31.1 | 220.0 | 303.7 | 7.14 ± 1.56 |
| `new_find google-chrome` | 540.5 ± 65.3 | 482.5 | 675.0 | 15.72 ± 3.38 |
| `new_glob nonexistent` | 37.6 ± 2.4 | 34.3 | 42.9 | 1.09 ± 0.21 |
| `old_find nonexistent` | 440.2 ± 42.3 | 383.4 | 499.3 | 12.80 ± 2.59 |
| `new_find nonexistent` | 553.7 ± 44.2 | 503.7 | 619.8 | 16.10 ± 3.14 |

Hyperfine summary:
```
  'old_find umka' ran
    1.05 ± 0.27 times faster than 'new_glob umka'
    1.07 ± 0.24 times faster than 'new_glob endgame'
    1.08 ± 0.30 times faster than 'new_glob google-chrome'
    1.09 ± 0.21 times faster than 'new_glob nonexistent'
    1.15 ± 0.25 times faster than 'new_find umka'
    1.19 ± 0.33 times faster than 'new_glob facter'
    7.14 ± 1.56 times faster than 'old_find google-chrome'
   10.08 ± 1.82 times faster than 'old_find facter'
   10.29 ± 1.85 times faster than 'old_find endgame'
   12.80 ± 2.59 times faster than 'old_find nonexistent'
   13.82 ± 2.48 times faster than 'new_find facter'
   15.07 ± 2.89 times faster than 'new_find endgame'
   15.72 ± 3.38 times faster than 'new_find google-chrome'
   16.10 ± 3.14 times faster than 'new_find nonexistent'
```

-----
To add a little stress, I created 30,000 additional formula files, in a set of 100 taps as follows:
```
touch test/homebrew-tap{0..99}/Formula/f{0.99}.rb
touch test/homebrew-tap{0..99}/HomebrewFormula/f{0.99}_hf.rb
touch test/homebrew-tap{0..99}/f{0.99}_root.rb
```
The results tell basically the same story:
| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `new_glob f42` | 49.9 ± 10.6 | 40.4 | 76.6 | 1.00 |
| `old_find f42` | 404.1 ± 24.3 | 371.9 | 446.3 | 8.10 ± 1.79 |
| `new_find f42` | 548.0 ± 40.3 | 508.5 | 611.4 | 10.98 ± 2.47 |
| `new_glob f42_hf` | 52.7 ± 10.1 | 43.6 | 80.0 | 1.06 ± 0.30 |
| `old_find f42_hf` | 404.1 ± 23.3 | 376.8 | 432.6 | 8.10 ± 1.78 |
| `new_find f42_hf` | 575.3 ± 60.7 | 514.4 | 680.9 | 11.53 ± 2.74 |
| `new_glob f42_root` | 221.4 ± 20.0 | 197.6 | 246.5 | 4.44 ± 1.02 |
| `old_find f42_root` | 440.0 ± 30.5 | 393.7 | 478.0 | 8.82 ± 1.97 |
| `new_find f42_root` | 579.4 ± 28.4 | 528.9 | 636.0 | 11.61 ± 2.53 |

Hyperfine summary:
```
  'new_glob f42' ran
    1.06 ± 0.30 times faster than 'new_glob f42_hf'
    4.44 ± 1.02 times faster than 'new_glob f42_root'
    8.10 ± 1.79 times faster than 'old_find f42'
    8.10 ± 1.78 times faster than 'old_find f42_hf'
    8.82 ± 1.97 times faster than 'old_find f42_root'
   10.98 ± 2.47 times faster than 'new_find f42'
   11.53 ± 2.74 times faster than 'new_find f42_hf'
   11.61 ± 2.53 times faster than 'new_find f42_root'
```

So bash globbing seems to hit the fast/consistent/correct trifecta.